### DISCUSSION TITLE
vendor/filament/filament/src/Resources/Resource.php:336

### DISCUSSION THREAD

**Discussion by shiroamada** [filamentphp/filament#6167] [General]
I am new to Filament,

My Laravel Project My User Model is park under
`app\User.php`

But I read the source code it is seem to hardcoded to this`->prepend('App\\Models\\');`
Is there any way I can change it? instead of move it under `App\\Models\\` ?

**Comment by danharrin** [comment]
Create the resource then just change the import at the top of the resource file to the correct model

**Comment by shiroamada** [comment]
I am using `phpsa/filament-authentication` packages, I able to navigate `permissions` and `roles`, but when I click Users, error prompt.

I edit `vendor/filament/filament/src/Resources/Resource.php:336` to `->prepend('App\\');`, it works.

What I have done
I run `php artisan make:filament-resource User` 
`app/Filament/Resources/UserResource.php` was created.

replace
`use App\Models\User;` to `use App\User;`

added 

```
public static function getModel(): string
    {
        return static::$model ?? (string) Str::of(class_basename(static::class))
            ->beforeLast('Resource')
            ->prepend('App\\');
    }
```

Still not working when I click User link. 
Maybe can help to guide how to edit correctly. I am new in filament.
Thanks

**Reply by danharrin** [reply]
What error

**Reply by shiroamada** [reply]
[Class "App\Models\User" not found](http://installer-biolux.test/super-admin/users#top)

**Reply by danharrin** [reply]
Search your code for the incorrect namespace and replace it

**Reply by shiroamada** [reply]
I create a new laravel 10, then install filament to test it, it works ok until I composer require for `phpsa/filament-authentication`. It prompt "App\Models\User" not found again.
I think I just move the User model back to App\Models\User to fix this issue.
Thanks @danharrin
